[HotFix] - v0.4.1

chengoconnell opened the pull request:
Hotfix for production to prevent users from creating fields with empty API ID.

chengoconnell committed:
feature(): bumping version for hotfix field id

chengoconnell committed:
Publish

 - @slicemachine/client@1.1.11-alpha.3
 - @slicemachine/core@1.1.11-alpha.3
 - @slicemachine/init@1.1.11-alpha.4
 - slice-machine-ui@0.4.1-alpha.8

chengoconnell committed:
feature(): error if no field id

chengoconnell committed:
Publish

 - slice-machine-ui@0.4.1-alpha.9

sintinilorenzo commented on packages/slice-machine/lib/forms/defaults.ts:
```diff
@@ -9,6 +9,9 @@ export const validateId = ({
   initialId: string;
 }) => {
+  if (!value) {
+    return "Field is required";
```
This might be a bit inconsistent with the error message from the label missing, which says `This field is required`. There are the errors throughout the app, may be worth using the same everywhere since it's a small change?
<img width="495" alt="image" src="https://user-images.githubusercontent.com/69144671/178299364-1ea02b1b-c82e-42f1-a5fd-467833fb1e3f.png">
<img width="431" alt="image" src="https://user-images.githubusercontent.com/69144671/178299834-db2b3ff3-20c3-42c7-b55f-3604f85190bf.png">

Heyrows committed:
On start, crash if models are not supported anymore and give information on how to fix them.

Heyrows committed:
Add retro compatibility and prevent new fields to have dash characters

Heyrows committed:
add tests for API ID regexes

Heyrows committed:
correct missing babel/runtime on core

Heyrows committed:
add test for validating models

Heyrows committed:
fix custom type ID for e2e test custom type creation

Heyrows committed:
fix new test

Heyrows committed:
fix wording & e2e test

Heyrows committed:
fix wording

Heyrows committed:
change slugify of custom type name according to API ID regex

Heyrows committed:
fix slugify tests

Heyrows committed:
Merge pull request #520 from prismicio/ac/hotfix-crash

[Hotfix] - Validate models on start & display information if models are wrong.

Heyrows committed:
Publish

 - @slicemachine/client@1.1.11-alpha.4
 - @slicemachine/core@1.1.11-alpha.4
 - @slicemachine/init@1.1.11-alpha.5
 - slice-machine-ui@0.4.1-alpha.10

Heyrows committed:
check for wrong slice ID on start of slice machine

Heyrows committed:
modify wording after Angelo's feedback

Heyrows committed:
Merge pull request #524 from prismicio/ac/hotfix-sliceid

[Hotfix] - add check on Slice ID

Heyrows committed:
Publish

 - slice-machine-ui@0.4.1-alpha.11
